packages/odyssey-react-mui/src/DataTable/DataTablePagination.tsx

```diff
@@ -0,0 +1,274 @@
+  // is generic, while plain KeyboardEvent is not. We need this generic so we can specify the HTMLInputElement,
+  // which allows us to use currentTarget.value
+  const handlePageSubmit = useCallback(
+    (event: React.KeyboardEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+      if (event.key === "Enter") {
+        setPagination({
+          pageIndex: parseInt(event.currentTarget.value),
+          pageSize: rowsPerPage,
+        });
+      }
+    },
+    [rowsPerPage, setPagination],
+  );
+
+  const handleRowsPerPageSubmit = useCallback(
+    (event: React.KeyboardEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+      if (event.key === "Enter") {
+        setPagination({
+          pageIndex: page,
+          pageSize: parseInt(event.currentTarget.value),
+        });
+      }
+    },
+    [page, setPagination],
+  );
+
+  const handleLoadMore = useCallback(() => {
+    setPagination({
+      pageIndex: 1,
+      pageSize: rowsPerPage + initialRowsPerPage.current,
+    });
+  }, [rowsPerPage, setPagination]);
+
+  const loadMoreIsDisabled = useMemo(() => {
+    return totalRows ? rowsPerPage >= totalRows : false;
+  }, [rowsPerPage, totalRows]);
+
+  return variant === "paged" ? (
+    <PaginationContainer>
+      <PaginationSegment odysseyDesignTokens={odysseyDesignTokens}>
+        <Box>
+          <Paragraph component="span" color="textSecondary">
+            {t("table.pagination.rowsperpage")}
+          </Paragraph>
+          <PaginationInput
+            odysseyDesignTokens={odysseyDesignTokens}
+            type="number"
+            value={rowsPerPage}
+            onChange={(event) => setRowsPerPage(parseInt(event.target.value))}
+            onBlur={() => handlePaginationChange()}
+            onKeyDown={(event) => handleRowsPerPageSubmit(event)}
+            disabled={isDisabled}
+            inputProps={{
+              "aria-label": t("table.pagination.rowsperpage"),
+            }}
+          />
+        </Box>
+        <Paragraph component="span" color="textSecondary">
+          {totalRows ? (
+            <Trans
+              i18nKey="table.pagination.rowswithtotal"
+              values={{ firstRow, lastRow, totalRows }}
+            />
+          ) : (
+            <Trans
+              i18nKey="table.pagination.rowswithouttotal"
+              values={{ firstRow, lastRow }}
+            />
+          )}
+        </Paragraph>
+      </PaginationSegment>
+
+      <PaginationSegment odysseyDesignTokens={odysseyDesignTokens}>
+        {totalRows && (
+          <Box>
+            <Paragraph component="span" color="textSecondary">
+              {t("table.pagination.page")}
+            </Paragraph>
+            <PaginationInput
+              odysseyDesignTokens={odysseyDesignTokens}
+              type="number"
+              value={page}
+              onChange={(event) => setPage(parseInt(event.target.value))}
```

Comment:
This can cause re-renders. Consider changing this to:
```
const onChangeHander = useCallback((event) => setPage(parseInt(event.target.value)), []);
```
Note: TS / ESLint may want you to type the event and you can get that from hovering over `onChange` above in your IDE

```
onChange={onChangeHandler}
```